staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -265,18 +460,196 @@ func (q *delayingType) waitingLoop() {
 }
 
 // insert adds the entry to the priority queue, or updates the readyAt if it already exists in the queue
-func insert(q *waitForPriorityQueue, knownEntries map[t]*waitFor, entry *waitFor) {
+// It returns a boolean to indicate that the entry was inserted or not.
+func insert(q *waitForPriorityQueue, knownEntries map[t]*waitFor, entry *waitFor) bool {
 	// if the entry already exists, update the time only if it would cause the item to be queued sooner
```

Comment:
Can you explain what you mean - it still looks true to me.  Maybe I should have change priority to 'waiting'

Reply:
maybe something like this?

```suggestion
	// if the entry already exists, update the time according to the entry.options.WhenWaiting
```


and also

```
// insert adds the entry to the 'waiting' queue, or updates the readyAt according to the entry.options.WhenWaiting if it already exists in the queue
```
since we also have an option `TakeExisting` that does nothing


Or just document everything above  the function and remove the `// if the entry already exists` comment?